samcli/lib/delete/cfn_utils.py

```diff
@@ -20,8 +26,21 @@ def has_stack(self, stack_name: str) -> bool:
         """
         Checks if a CloudFormation stack with given name exists
 
-        :param stack_name: Name or ID of the stack
-        :return: True if stack exists. False otherwise
+        Parameters
+        ----------
+        stack_name: str
+            Name or ID of the stack
+
+        Returns
+        -------
+        bool
+            True if stack exists. False otherwise
+
+        Raises
+        ------
+        DeleteFailedError
```

Comment:
nit: the exception name looks weird to be raised from `has_stack` function.

Reply:
can we change the function name to be something like `can_delete_stack`

Reply:
Sure, we can update this method to better represent what could happen